src/test/java/com/puppycrawl/tools/checkstyle/XpathFileGeneratorAstFilterTest.java

```diff
@@ -0,0 +1,142 @@
+////////////////////////////////////////////////////////////////////////////////
+// checkstyle: Checks Java source code for adherence to a set of rules.
+// Copyright (C) 2001-2018 the original author or authors.
+//
+// This library is free software; you can redistribute it and/or
+// modify it under the terms of the GNU Lesser General Public
+// License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+//
+// This library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+// Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public
+// License along with this library; if not, write to the Free Software
+// Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+////////////////////////////////////////////////////////////////////////////////
+
+package com.puppycrawl.tools.checkstyle;
+
+import static org.junit.Assert.assertArrayEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+import java.lang.reflect.Field;
+import java.lang.reflect.Modifier;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+import org.junit.Before;
+import org.junit.Test;
+
+import com.puppycrawl.tools.checkstyle.api.TokenTypes;
+import com.puppycrawl.tools.checkstyle.checks.blocks.LeftCurlyCheck;
+import com.puppycrawl.tools.checkstyle.checks.coding.RequireThisCheck;
+import com.puppycrawl.tools.checkstyle.checks.whitespace.MethodParamPadCheck;
+
+public class XpathFileGeneratorAstFilterTest extends AbstractFileGeneratorTestSupport {
+
+    private Field eventsField;
+
+    private Field queriesField;
+
+    @Before
+    public void setUp() throws Exception {
+        eventsField = XpathFileGeneratorAstFilter.class.getDeclaredField("EVENTS");
+        eventsField.setAccessible(true);
+
+        queriesField = XpathFileGeneratorAstFilter.class.getDeclaredField("QUERIES");
+        queriesField.setAccessible(true);
```

Comment:
Since `EVENTS` and `QUERIES` are private static final fields and have no getters, how can I check their values?

> Do you have problems to make this Test by some Input file and predictive result ?

I have input and predictive result in `XpathFileGeneratorAuditListenerTest`, there I check for generated suppressions file, here I wanted to check for generated xpath queries, 
but I think all functionality of the `XpathFileGeneratorAstFilter` is covered in `XpathFileGeneratorAuditListenerTest` and `MainTest` so I can just remove these reflection test methods

Reply:
@timurt ,
ALL tests that require magic of reflection or fancy mocks should have comment that explain reason of hack.
if tests are not required please remove them.

PR become too big, and we need to merge it, so if this reflection based test is do some coverage that missed in other tests, please put comment "till <Issue>" and open new issue on this to complete it right after this PR, diff will be much smaller , easier to resolve.